### ISSUE TITLE
Create a collaboration-friendly pipeline solution for Terraform

### ISSUE THREAD

**Issue by mischavandenburg** [mischavandenburg/kubecraft#25]
Eventually we can use both Bicep and Terraform.

This issue can be copied for Bicep.

We need a GitHub Actions workflow that allows deployment of Terraform code to Azure.

Test if we can have a setup that supports forked repositories.

Someone who forks should be able to point the github actions to their own Azure subscription for testing.

**Comment by Lucioschenkel** [comment]
@mischavandenburg I'd like to work on this. **Question:** Do we want to support multiple environments from the same pipeline, and what events are we going to support? e.g. manual trigger (`workflow_dispatch`), `push` to a specific branch pattern, etc.

**Comment by mischavandenburg** [comment]
Great to hear @Lucioschenkel ! For now we can just have one environment, but in the future we will at least have staging and production.

Please let the trigger run when the main brench receives commits in the terraform directory

**Comment by mischavandenburg** [comment]
@Lucioschenkel , while I was running the new Terraform code for the VM, it occured to me that we also need a way to store the state file.

Are you OK with incorporating this into the pipeline as well?

https://learn.microsoft.com/en-us/azure/developer/terraform/store-state-in-azure-storage?tabs=azure-cli

Feel free to ask for help in the chat, I'm sure there are others who would like to help

**Comment by Lucioschenkel** [comment]
@mischavandenburg Yep, no problem. That's great because I never worked with Terraform and Azure before, so I'm learning a bunch of new stuff 😄
